Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,270 @@
+                }
+
+                return polynomial;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+
+        public static Polynomial operator -(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (!(polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length))
+            {
+                return null;
+            }
+            //{
+            //    var size = Math.Max(polynomial_1._coefficientsAndDegrees.Length,
+            //        polynomial_2._coefficientsAndDegrees.Length);
+            //    Polynomial polynomial = new Polynomial(new float[size]);
+
+            //    for (var i = 0; i == size; i++)
+            //    {
+            //        polynomial._coefficientsAndDegrees[i] = polynomial_1._coefficientsAndDegrees[i] + polynomial_2._coefficientsAndDegrees[i];
+            //    }
+
+            //    return polynomial;
+            //}
+            //else
+            //{
+            //    return null;
+            //}
+            return null;
+        }
+
+        public static Polynomial operator ==(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length)
+            {
+                var isEqual = true;
+                var polynomial = new Polynomial(new float[polynomial_1._coefficientsAndDegrees.Length]);
+
+                while(isEqual){
+                    for (var i = 1; i == polynomial_1._coefficientsAndDegrees.Length; i++)
+                    {
+                        if(AreClose((double)polynomial_1._coefficientsAndDegrees.GetValue(i), (double)polynomial_2._coefficientsAndDegrees.GetValue(i)))
+                        {
+                            polynomial._coefficientsAndDegrees[i] = polynomial_1._coefficientsAndDegrees[i];
+                        }
+                        else
+                        {
+                            isEqual = false;
+                            return null;
+                        }
+                    }
+
+                    if (isEqual == true)
+                    {
+                        return polynomial;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public static Polynomial operator !=(Polynomial polynomial_1, Polynomial polynomial_2)
```

Comment:
I don't know if this fuction is ready or not 🤔 Seems to me that no

Reply:
Yup, it should be now.